### PR TITLE
Introduce `__addError()` Function to Add an `ErrorDetail` into the `errors` Field of a GraphQL Response

### PR DESCRIPTION
**Pull request by Ishad-M-I-M**
## Purpose

Fixes: [#4479](https://github.com/ballerina-platform/ballerina-standard-library/issues/4479)

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [x] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: docs/spec/spec.md [hunk 1]
```diff
@@ -1608,6 +1609,45 @@ To resolve the value of a field, the `resolve()` method can be used. This requir
 public isolated function resolve(graphql:Field ‘field) returns anydata;
 ```
 
+### 8.6 Adding ErrorDetail to the Context
+
+To add an `ErrorDetail` to the context, the `addError()` method can be used. This method requires a `ErrorDetail` object as the parameter.
```
Comment (MohamedSabthar) [inline docs/spec/spec.md:1614]: I believe it's important to clarify how including an error in the context can affect the overall GraphQL response.

Comment (Ishad-M-I-M) [reply docs/spec/spec.md:1614]: Overall Graphql response will be affected as in this example.
https://github.com/ballerina-platform/module-ballerina-graphql/blob/master/docs/spec/spec.md#example-sample-response-for-error-returned-from-a-nullable-field
I'll update it.

File: ballerina-tests/tests/resources/documents/null_value_for_nullable_input_object_with_error_added_to_context.graphql [hunk 2]
```diff
@@ -0,0 +1,3 @@
+{
+    name(id: null)
+}
```
Comment (ThisaruGuruge) [inline ballerina-tests/tests/resources/documents/null_value_for_nullable_input_object_with_error_added_to_context.graphql:3]:
```suggestion
}

```

File: ballerina-tests/tests/resources/expected_results/null_value_for_nullable_input_object_with_error_added_to_context.json [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+    "errors": [
+        {
+            "message": "Data not found",
+            "path": ["name"]
+        }
+    ],
+    "data" : {
+        "name" : null
+    }
+}
```
Comment (ThisaruGuruge) [inline ballerina-tests/tests/resources/expected_results/null_value_for_nullable_input_object_with_error_added_to_context.json:11]: Check the IDE settings whether it adds an empty line at the end of each file.

```suggestion
}

```

**Review by ThisaruGuruge — APPROVED** [review]

**Review by ThisaruGuruge — COMMENTED** [review]
Shall we update the Changelog?

**Pull request merged**